openwisp_controller/config/admin.py

```diff
@@ -52,6 +54,25 @@
     from django.contrib.admin import ModelAdmin
 
 
+def get_system_context_output(obj):
+    if hasattr(obj, 'device'):
+        system_context = obj.device.get_system_context()
+    else:
+        system_context = obj.get_system_context()
```

Comment:
this if is ugly, is it necessary? Can we find a way to avoid it?

Why are you calling it? Config, Template and Vpn should have a `get_system_context`, so you should just call `system_context = obj.get_system_context()`

Reply:
I did this because my understanding of how the device and config works was wrong. thank you for point this out.